skops/hub_utils/_hf_hub.py

```diff
@@ -96,28 +99,79 @@ def _get_example_input(data):
         # pandas is not installed, the data cannot be a pandas DataFrame
         pass
 
-    # here we convert the first three rows of the numpy array to a dict of lists
+    # here we convert the first three rows of `data` to a dict of lists
     # to be stored in the config file
-    if isinstance(data, np.ndarray):
-        return {f"x{x}": data[:3, x].tolist() for x in range(data.shape[1])}
+    if isinstance(data, (np.ndarray, list, tuple)):
+        data_slice = data[:3]
+        # This will raise a ValueError if the array is not 2D
+        data_slice_array = check_array(data_slice, ensure_2d=True)
+        return {
+            f"x{x}": data_slice_array[:, x].tolist()
+            for x in range(data_slice_array.shape[1])
+        }
+
+    raise ValueError(
+        "The data is not a pandas.DataFrame, a 2D numpy.ndarray or a "
+        "list/tuple that can be converted to a 2D numpy.ndarray."
+    )
+
+
+def _get_example_input_from_text_data(data: Sequence[str]):
+    """Returns the example input of a model for a text task.
+
+    The input is converted into a dictionary which is then stored in the config
+    file.
+
+    Parameters
+    ----------
+    data: Sequence[str]
+        A sequence of strings. The first 3 elements are used as example input.
+
+    Returns
+    -------
+    example_input: dict of lists
+        The example input of the model as accepted by Hugging Face's backend.
+    """
+
+    def _head(data, n):
+        def is_subscriptable(data):
+            return hasattr(data, "__getitem__")
+
+        if is_subscriptable(data):
+            return data[:n]
+
+        return list(itertools.islice(data, n))
 
-    raise ValueError("The data is not a pandas.DataFrame or a numpy.ndarray.")
+    def _is_sequence_of_strings(data):
+        return not isinstance(data, str) and all(isinstance(x, str) for x in data)
+
+    error_message = "The data needs to be a sequence of strings."
```

Comment:
There isn't really a need to assign the error message here. It is better to move it into the `except` part below.

Reply:
The reason I assigned the error message there is because otherwise I would have to duplicate the same string in the `else` and `except` branches, like so:
```python
try:
    data_head = _head(data, n=3)
    if _is_sequence_of_strings(data_head):
        return {"data": data_head}
    else:
        raise ValueError("The data needs to be a sequence of strings.")
except TypeError as e:
    raise ValueError("The data needs to be a sequence of strings.") from e
```
Another way to avoid this would be to raise a `TypeError` inside the `else` branch, although that would feel pretty hacky.
Do you think it's fine to duplicate the error message?

Reply:
Sorry, I have missed that it's used at two places.